src/main/com/fulcrologic/fulcro/algorithms/timbre_support.cljc

```diff
@@ -0,0 +1,100 @@
+(ns com.fulcrologic.fulcro.algorithms.timbre-support
+  "
+  Logging helpers to make js console logging more readable. The recommended use of these functions is as follows:
+
+  - Make sure you're using Binaryage devtools (on classpath. shadow-cljs will auto-add it when detected).
+  - IMPORTANT: Enable custom formatters in console settings for Chrome. This will print cljs data as cljs (instead of raw js).
+  - Make a development preload cljs file, and tell shadow-cljs to preload it.
+  - In the preload file, add something like this:
+
+  ```
+  (ns app.development-preload
+    (:require
+      [taoensso.timbre :as log]
+      [com.fulcrologic.fulcro.algorithms.timbre-support :refer [console-appender prefix-output-fn]))
+
+  (log/set-level! :debug)
+  (log/merge-config! {:output-fn prefix-output-fn
+                      :appenders {:console (console-appender)}})
+  ```
+
+  and you'll get much more readable error messages in the js console.
+
+  NOTE: when logging errors, be sure to log the exception first. This is documented in timbre, but easy to miss:
+
+  ```
+  (try
+    ...
+    (catch :default ex
+      (log/error ex ...))
+  ```
+
+  See the development_preload.cljs and shadow-cljs.edn files in the latest Fulcro 3 template for an example.
+  "
+(:require
+  [clojure.string :as str]
+  [taoensso.encore :as encore]
+  [taoensso.timbre :as log]))
+
+;; Taken mostly from timbre itself.  Modified to output better results from ex-info exceptions (e.g. to improve expound
+;; experience)
+#?(:cljs
```

Comment:
In general put these in the body in case it is used in a CLJC file, so that the function is at least defined. Then note in docstring that it is only implemented for cljs.